src/microbe_stage/ChunkSpawner.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using Godot;
+
+/// <summary>
+///   Spawns chunks of a specific type
+/// </summary>
+public class ChunkSpawner
+{
+    private readonly PackedScene chunkScene;
+    private readonly Random random;
+    private readonly CompoundCloudSystem cloudSystem;
+
+    public ChunkSpawner(CompoundCloudSystem cloudSystem, Random random)
+    {
+        this.cloudSystem = cloudSystem;
+        this.random = random;
+        chunkScene = LoadChunkScene();
+    }
+
+    public static PackedScene LoadChunkScene()
+    {
+        return GD.Load<PackedScene>("res://src/microbe_stage/FloatingChunk.tscn");
+    }
+
+    public static ISpawned SpawnChunk(ChunkConfiguration chunkType, Vector3 location, Node worldNode,
```

Comment:
Why did you keep the separate class for this but not for the agentprojectile? I think for consistency one of the two approaches should be used, and not both.

Reply:
Each SpawnItem points to their associated spawner so they can, well spawn. Also, a dying microbe needs to run the SpawnChunk code as well as the ChunkItem. That was my reason why I kept this but moved the agentprojectile.

Reply:
You can still reference static methods no matter which class they are in? Anyway I think only one of the approaches should be used because someone new reading the code may be confused by the same operation being in different places for different classes.